server/chat-commands/info.ts

```diff
@@ -2610,13 +2610,16 @@ export const commands: ChatCommands = {
 		}
 
 		this.checkBroadcast(true, '!code');
-
-		const code = Chat.getReadmoreCodeBlock(target);
 		this.runBroadcast(true);
-		if (this.broadcasting) {
-			return `/raw <div class="infobox">${code}</div>`;
+
+		const isPMOrPersonalRoom = this.room?.settings.isPersonal !== false;
+
+		if (this.broadcasting && isPMOrPersonalRoom) {
+			target = Chat.filter(this, target, user, room, connection, this.pmTarget)!;
+			if (!target) return this.errorReply(`Invalid code.`);
```

Comment:
Am I misunderstanding this code? I think you meant:

```suggestion
		if (this.broadcasting) {
			if (isPMOrPersonalRoom) {
				target = Chat.filter(this, target, user, room, connection, this.pmTarget)!;
				if (!target) return this.errorReply(`Invalid code.`);
			}
```

Reply:
yea i wasnt thinking straight